geode-apis-compatible-with-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/sortedset/ZRemRangeByRankDUnitTest.java

```diff
@@ -0,0 +1,244 @@
+      throws Exception {
+    Map<String, Double> memberScoreMap = makeMemberScoreMap();
+
+    jedis.zadd(KEY, memberScoreMap);
+    verifyDataExists(memberScoreMap);
+
+    String firstMember = String.join("", jedis.zrange(KEY, 0, 0));
+    memberScoreMap.remove(firstMember);
+
+    Future<Void> future1 = executor.submit(() -> stopNodeWithPrimaryBucketOfTheKey(true));
+    Future<Void> future2 = executor.submit(this::removeAllButFirstEntry);
+
+    future1.get();
+    future2.get();
+
+    await().until(() -> verifyDataDoesNotExist(memberScoreMap));
+    assertThat(jedis.zrank(KEY, firstMember)).isZero();
+    assertThat(jedis.exists(KEY)).isTrue();
+  }
+
+  private void verifyDataExists(Map<String, Double> memberScoreMap) {
+    for (String member : memberScoreMap.keySet()) {
+      Double score = jedis.zscore(KEY, member);
+      assertThat(score).isEqualTo(memberScoreMap.get(member));
+    }
+  }
+
+  private boolean verifyDataDoesNotExist(Map<String, Double> memberScoreMap) {
+    try {
+      for (String member : memberScoreMap.keySet()) {
+        Double score = jedis.zscore(KEY, member);
+        assertThat(score).isNull();
+      }
+    } catch (JedisClusterMaxAttemptsException e) {
+      return false;
+    }
+    return true;
+  }
+
+  private void doZRemRangeByRankOnMembers(int numberOfMemberToRemove, AtomicInteger total) {
+    long count = jedis.zremrangeByRank(KEY, 0, numberOfMemberToRemove - 1);
+    total.addAndGet((int) count);
+  }
+
+  private void doZRemRangeByRankOnMembersWithOffset(int numberOfMemberToRemove, int offset,
+      AtomicInteger total) {
+    long count = jedis.zremrangeByRank(KEY, offset, (offset + numberOfMemberToRemove) - 1);
+    total.addAndGet((int) count);
+  }
+
+  private void doZRemRangeByRankWithRetries(Map<String, Double> map) {
+    int maxRetryAttempts = 10;
+    int retryAttempts = 0;
+    while (!zRemRangeByRankWithRetries(map, retryAttempts, maxRetryAttempts)) {
+      retryAttempts++;
+    }
+  }
+
+  private boolean zRemRangeByRankWithRetries(Map<String, Double> map, int retries, int maxRetries) {
+    long removed;
+    try {
+      removed = jedis.zremrangeByRank(KEY, 0, -1);
+    } catch (JedisClusterMaxAttemptsException e) {
+      if (retries < maxRetries) {
+        return false;
+      }
+      throw e;
+    }
+    assertThat(removed).isEqualTo(map.size());
+    return true;
+  }
+
+  private void removeAllButFirstEntry() {
+    long removed = 0;
+    int rangeSize = 10;
+    await().until(isCrashing::get);
+    for (int i = 1; i < SET_SIZE; i += rangeSize) {
+      removed += jedis.zremrangeByRank(KEY, 1, rangeSize);
+    }
+    assertThat(removed).isEqualTo(SET_SIZE - 1);
+  }
+
+  private void stopNodeWithPrimaryBucketOfTheKey(boolean isCrash) {
+    boolean isPrimary;
+    for (MemberVM server : servers) {
+      isPrimary = server.invoke(ZRemRangeByRankDUnitTest::isPrimaryForKey);
+      if (isPrimary) {
+        if (isCrash) {
+          isCrashing.set(true);
```

Comment:
Not a big deal, but consider resetting this in `@Before` since future tests might rely on it.

Reply:
Good call